cli/Sources/TuistSupport/Xcode/XcodeController.swift

```diff
@@ -29,13 +29,26 @@ public final class XcodeController: XcodeControlling, @unchecked Sendable {
 
     /// Returns the selected Xcode. It uses xcode-select to determine
     /// the Xcode that is selected in the environment.
+    /// If `DEVELOPER_DIR` is set, it takes precedence over what
+    /// xcode-select returns.
     ///
     /// - Returns: Selected Xcode.
     /// - Throws: An error if it can't be obtained.
     public func selected() async throws -> Xcode {
         if let selectedXcode = selectedXcode.value {
             return selectedXcode
         } else {
+            if let environmentValue = Environment.current.variables["DEVELOPER_DIR"] {
+                // It's valid to either point DEVELOPER_DIR directly to Xcode.app/ - or
+                // to the developer content directory. See man xcode-select # ENVIRONMENT
+                var path = try AbsolutePath(validating: environmentValue)
+                if path.pathString.hasSuffix("Contents/Developer") {
+                    path = path.parentDirectory.parentDirectory
+                }
+                let value = try await Xcode.read(path: path)
+                selectedXcode.mutate { $0 = value }
+                return value
```

Comment:
this is common for both `if-else` branches. I'd move this logic after `if-else` and let the `if-else` branch handle only the different Xcode paths